presto-main/src/main/java/io/prestosql/security/ViewAccessControl.java

```diff
@@ -74,4 +78,14 @@ public void checkCanGrantExecuteFunctionPrivilege(SecurityContext context, Strin
     {
         return delegate.getColumnMasks(context, tableName, columnName, type);
     }
+
+    private void executeAsViewOwner(Consumer<AccessControl> consumer)
```

Comment:
This could take `Runnable` since the caller already has the delegate. Then this method can be static.

Reply:
I considered Runnable as well, however I did not find good method name.

`executeAsViewOwner(() -> delegate.checkCanCreateViewWithSelectFromColumns(context, tableName, columnNames));`
does not sound good, because it does not execute something as owner it just wraps it to owner error. 

`wrapWithOwnerException(() -> delegate.checkCanCreateViewWithSelectFromColumns(context, tableName, columnNames));` is not good because you can write whatever you want within runnable and it will be just wraped with exception. No relation between delegate(view owner) and exception at all.

`executeAsViewOwner(owner -> owner.checkCanCreateViewWithSelectFromColumns(context, tableName, columnNames));` I feel that this code somehow obliges me to utilize owner parameter. Once I use it - I have an exception related to this parameter.
Despite this version is a little bit longer I prefer it, unless you have better suggestion.

Reply:
> I considered Runnable as well, however I did not find good method name.


Let's call it `wrapAccessDeniedException`

Reply:
I still don't like it, but 2:1. Renamed.

Reply:
@electrum why do you prefer static over non static methods?

Reply:
static method is just a function while method is a method ;) 

 - Function does not requires polymorphism and it is much easier for compiler and runtime.
 - It is much easier to maintain that, one can easily extract this function out of this class and reuse it somewhere.
 - It is more readable, reader knows that this function won't modify or even access the object state.